src/utils/memory_size.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <multipass/exceptions/invalid_memory_size_exception.h>
+#include <multipass/memory_size.h>
+#include <multipass/utils.h>
+
+#include <fmt/format.h>
+
+namespace mp = multipass;
+
+namespace
+{
+constexpr auto kilo = 1024LL;
+constexpr auto mega = kilo * kilo;
+constexpr auto giga = mega * kilo;
+
+long long in_bytes(const std::string& mem_value)
+{
+    QRegExp matcher("^(\\d+)([KMG])?B?$", Qt::CaseInsensitive);
+
+    if (matcher.exactMatch(QString::fromStdString(mem_value)))
+    {
+        auto val = matcher.cap(1).toLongLong(); // value is in the second capture (1st one is the whole match)
+        const auto unit = matcher.cap(2);       // unit in the third capture (idem)
```

Comment:
This is standard for regex captures, not sure if the comments are necessary.

Reply:
I understand your point. I just commented what I thought I would forget ("document what you know when you know it" I read somewhere and caught in my ear). I suppose I have used regex groups as much as the next programmer, but that is still sparsely enough that I always end up forgetting these details. I wanted to ease it a bit on the reader and save myself (and others) from having to go consult the documentation when I read this code in a couple of months.

But I agree comments should add something and the code is hopefully clear enough. So let me know if you still think they are not helpful and I will remove it.

Reply:
It's ok, they can stay :)